resources/shared-db/src/postgres.rs

```diff
@@ -59,14 +60,30 @@ impl ResourceBuilder<sqlx::PgPool> for Postgres {
             DbOutput::Info(info) => info.connection_string_private(),
         };
 
-        let pool = sqlx::postgres::PgPoolOptions::new()
-            .min_connections(1)
-            .max_connections(5)
-            .connect(&connection_string)
-            .await
-            .map_err(CustomError::new)?;
-
-        Ok(pool)
+        let mut attempt = 0;
+        let mut timeout_in_seconds = 1;
+        let max_attempt = 3;
+         
+        loop {
+            match sqlx::postgres::PgPoolOptions::new()
+                .min_connections(1)
+                .max_connections(5)
+                .connect(&connection_string)
+                .await
+            {
+                Ok(pool) => return Ok(pool),
+                Err(err) => {
+                    if attempt >= max_attempt {
+                        Err(CustomError::new(err))?;
```

Comment:
Not a big fan of relying on `?` for returning that way, I would be happier with a proper `return`, but this means using the following form, and I'm not sure that's better.
```rust
return Err(<Error as From<CustomError>>::from(CustomError::new(err)));
```